photon-lib/src/main/scala/com/linkedin/photon/ml/optimization/LBFGSB.scala

```diff
@@ -0,0 +1,110 @@
+    tolerance: Double = LBFGSB.DEFAULT_TOLERANCE,
+    maxNumIterations: Int = LBFGSB.DEFAULT_MAX_ITER,
+    isTrackingState: Boolean = Optimizer.DEFAULT_TRACKING_STATE)
+  extends LBFGS(
+    normalizationContext,
+    numCorrections,
+    tolerance,
+    maxNumIterations,
+    isTrackingState) {
+
+  /**
+   * Under the hood, this adaptor uses an LBFGSB optimizer from Breeze to optimize functions.
+   * The DiffFunction is modified into a Breeze DiffFunction which the Breeze optimizer can understand.
+   */
+  //  protected class BreezeOptimization(
+  //      breezeDiffFunction: BreezeDiffFunction[DenseVector[Double]],
+  //      initCoefficients: Vector[Double])
+  //    extends BreezeOptimization(breezeDiffFunction.asInstanceOf[BreezeDiffFunction[Vector[Double]]], initCoefficients)
+  //
+  //  protected val BreezeOptimizer = new BreezeLBFGSB(lowerBounds, upperBounds, maxNumIterations, numCorrections, tolerance)
+  //
+  protected class BreezeBOptimization(
+      breezeDiffFunction: BreezeDiffFunction[DenseVector[Double]],
+      initCoefficients: Vector[Double])
+    extends BreezeOptimization(breezeDiffFunction.asInstanceOf[BreezeDiffFunction[Vector[Double]]], initCoefficients)
+
+  // Cast into FirstOrderMinimizer as for overriding breezeOptimizer in parent class LBFGS
+  override protected val breezeOptimizer = new BreezeLBFGSB(lowerBounds, upperBounds, maxNumIterations, numCorrections, tolerance).
+    asInstanceOf[FirstOrderMinimizer[Vector[Double], BreezeDiffFunction[Vector[Double]]]]
+
+  /**
+   * Initialize breeze optimization engine.
+   *
+   * @param objectiveFunction The objective function to be optimized
+   * @param initState         The initial state of the optimizer, prior to starting optimization
+   * @param data              The training data
+   */
+
+  override def init(objectiveFunction: DiffFunction, initState: OptimizerState)(data: objectiveFunction.Data): Unit = {
+    val breezeDiffFunction: BreezeDiffFunction[DenseVector[Double]] = new BreezeDiffFunction[DenseVector[Double]] {
+      override def calculate(x: DenseVector[Double]): (Double, DenseVector[Double]) = {
+        var convertedCoefficients: objectiveFunction.Coefficients = objectiveFunction.convertFromVector(x)
+        var result = objectiveFunction.calculate(data, convertedCoefficients, normalizationContext)
+        (result._1, result._2.toDenseVector)
+      }
+    }
+    breezeOptimization = new BreezeBOptimization(breezeDiffFunction, initState.coefficients)
+
+  }
+
+  /**
+   * Just reset the whole BreezeOptimization instance.
+   */
+  override def clearOptimizerInnerState(): Unit = {
+
+    super.clearOptimizerInnerState()
+    breezeOptimization = _: BreezeBOptimization
+  }
+}
+
+object LBFGSB {
+  // TODO: Default tolerance and max # iterations should be in GameParams or GLMOptimizationConfiguration defaults
+  val DEFAULT_MAX_ITER = 100
+  val DEFAULT_NUM_CORRECTIONS = 10
+  val DEFAULT_TOLERANCE = 1.0E-7
```

Comment:
You can replace calls to these parameters with the ones from `LBFGS`. It'll be better that way since there will be an error if we ever replace them in `LBFGS` as the TODO comment suggests.